src/main/webapp/app/exercise/result/result.utils.ts

```diff
@@ -257,13 +257,13 @@ export const evaluateTemplateStatus = (
  * Checks if only compilation was tested. This is the case, when a successful result is present with 0 of 0 passed tests
  * This could be because all test cases are only visible after the due date.
  */
-export const isOnlyCompilationTested = (result: Result | undefined, templateStatus: ResultTemplateStatus): boolean => {
+export const isOnlyCompilationTested = (result: Result | undefined, participation: Participation, templateStatus: ResultTemplateStatus): boolean => {
     const zeroTests = !result?.testCaseCount;
-    const isProgrammingExercise: boolean = result?.submission?.participation?.exercise?.type === ExerciseType.PROGRAMMING;
+    const isProgrammingExercise: boolean = participation?.exercise?.type === ExerciseType.PROGRAMMING;
     return (
         templateStatus !== ResultTemplateStatus.NO_RESULT &&
         templateStatus !== ResultTemplateStatus.IS_BUILDING &&
-        !isBuildFailed(result?.submission) &&
+        !isBuildFailed(participation?.submissions?.[0]) &&
```

Comment:
We do not check the length before accessing the 0th index - this can cause issues.

I do also think that this should be a method in `Participation` instead - probably `getLatesteSubmission()`?

That would make the code much better readable and we can have the length checks once in the method instead of repeating it over and over again